mlflow/store/artifact/hdfs_artifact_repo.py

```diff
@@ -145,7 +144,12 @@ def download_artifacts(self, artifact_path, dst_path=None):
             return local_dir
 
     def _download_file(self, remote_file_path, local_path):
-        raise MlflowException('This is not implemented. Should never be called.')
+        _, _, path = _resolve_connection_params(remote_file_path)
+        if path.endswith('/'):
+            path = path[:-1]
+        with hdfs_system(host=self.host, port=self.port) as hdfs:
+            local_path = os.path.join(local_path, os.path.normpath(path.split('/')[-1]))
+            _download_hdfs_file(hdfs, path, local_path)
```

Comment:
downloading a file is already included in `download_artifacts` function. Is this doing any special handling other than handling the trailing `/`. Do you need to do that?

Maybe we can revert this functionality and make `download_artifacts` work for all cases.